packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -250,6 +252,16 @@ export default class OperatorModeStateService extends Service {
     }
   }
 
+  replaceCodePath(codePath: URL | null) {
+    // replace history explicitly
+    this.state.codePath = codePath;
+    this.router.replaceWith('card', {
+      queryParams: {
+        operatorModeState: this.serialize(),
+      },
+    });
```

Comment:
Perhaps `schedulePersist()` is needed here

Reply:
we weren't super sure about that. i vaguely recall schedule persist was needed for the stack item rendering. not sure we need that for code mode. maybe @lukemelia has thoughts here?

Reply:
`schedulePersist` is used to address a race condition when there are many state changes in a loop:

A comment that we have for that method:

> // When multiple stack manipulations are bunched together in a loop, for example when closing multiple cards in a loop,
    // we get into a async race condition where the change to cardController.operatorModeState will reload the route and
    // restore the state from the query param in a way that is out of sync with the state in the service. To avoid this,
    // we do the change to the query param only after all modifications to the state have been rendered.
    
> not sure we need that for code mode

Agree. I think we do not need it here